temporian/core/operators/test/test_arithmetic_multi_index.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from absl.testing import absltest
+
+from temporian.implementation.numpy.data.io import event_set
+
+from temporian.test.utils import f32, f64, assertOperatorResult
+
+# store ids
+TRYOLABS_SHOP = 42
+GOOGLE_SHOP = 101
+# product ids
+MATE_ID = 1
+BOOK_ID = 2
+PIXEL_ID = 3
+
+class ArithmeticMultiIndexNumpyImplementationTest(absltest.TestCase):
+    """Test numpy implementation of all arithmetic operators,
+    but using a two-level index and disordered rows."""
+
+    def setUp(self):
+        # 2 index columns, 2 feature columns (float64 and float32)
+        self.evset_1 = event_set(
+            timestamps=[0, 1, 2, 1, 1, 2, 0, 1],
+            features={
+                "store_id": [TRYOLABS_SHOP] * 4 + [GOOGLE_SHOP] * 4,
+                "product_id": [MATE_ID] * 3 + [BOOK_ID] * 3 + [PIXEL_ID] * 2,
+                "f1": f64([-14.0, 15.0, 16.0, 10.0, 0.0, 8.0, 3.0, 4.0]),
+                "f2": f32([1.0, 2.0, 3, 4, 5, 6, 7, 8]),
+            },
+            indexes=["store_id", "product_id"]
+        )
+
+        # 2 index columns, 2 feature columns (float64 and float32)
+        self.evset_2 = event_set(
+            timestamps=[0, 1, 2, 1, 1, 2, 0, 1],
+            features={
+                "store_id": [TRYOLABS_SHOP] * 4 + [GOOGLE_SHOP] * 4,
+                "product_id": [MATE_ID] * 3 + [BOOK_ID] * 3 + [PIXEL_ID] * 2,
+                "f3": f64([4.5, -5.5, 16, 0, 3, -8, 3, 5]),
+                "f4": f32([5, 3, 1, 2, -8, 2, 4, 3]),
+            },
+            indexes=["store_id", "product_id"],
+            same_sampling_as=self.evset_1
+        )
```

Comment:
same here, I'd try and simplify as much as possible (remove store var names, remove list multiplications, halve the number of events?). Some of the olds tests are unnecessarily complex. See for example how I simplified `rename`'s tests here when compared to the old ones: https://github.com/google/temporian/pull/286/files#diff-a9119b805f6d05d7d9950728b1aae5e51ac5ce1a6cb5a0c1a917de1673588bae

Reply:
Agreed. didn't reduce the number of events though, because they are useful to test different scenarios in all arithmetic ops.